Improve unittest cmake and gitignore

aaronj0 opened the pull request:
# Description

General improvements

## Type of change

Please tick all options which are relevant.

- [X] Bug fix
- [ ] New feature
- [ ] Requires documentation updates

mcbarton commented:
@aaronj0 @vgvassilev Almost certainly nothing to do with this PR, but any idea why the coverage report comment is not coming up for this PR? I can see from the ci that the repo is created and uploaded. It is appearing on other recent PRs so a little confused.

vgvassilev commented:
No clue, maybe some temporary outage?

aaronj0 commented:
> @aaronj0 @vgvassilev Almost certainly nothing to do with this PR, but any idea why the coverage report comment is not coming up for this PR? I can see from the ci that the repo is created and uploaded. It is appearing on other recent PRs so a little confused.

@mcbarton not sure why, I just rebased, lets see if codecov runs now. Otherwise can you take a look at the PR. I have updated the valgrind command since the CppInterOp tests path has been updated
